lib/src/main/java/com/metaplex/lib/experimental/serialization/serializers/rpc/RpcSerializers.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * RpcSerializers
+ * Metaplex
+ * 
+ * Created by Marco Martinez on 7/22/2022
+ * Copyright © Daxko, LLC 2022
+ */
+
+package com.metaplex.lib.experimental.serialization.serializers.rpc
+
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+class JsonRpcSerializer2<D, ED>(val dataSerializer: KSerializer<D>,
+                                errorDataSerializer: KSerializer<ED>) : KSerializer<RpcResponse> {
```

Comment:
its annoying and potentially confusing that we are forced to define the returned error type, need to find a way to abstract this away and make it an optional advanced configuration if it's needed.

Alternatively we could simply ignore the additional error data that _might_ be present if we don't need to care about it.

Reply:
I think I will leave this as-is for now, and refactor it after I have used it a bit more (need get a sense of the devex to inform my decision here)